src/experiments.js

```diff
@@ -226,8 +222,63 @@ Test.registerGoal( 'click', ( element, record ) => {
 	} );
 }, [ 'a' ] );
 
+class ExperienceBlock extends HTMLElement {
+
+	get clientId() {
+		return this.getAttribute( 'client-id' );
+	}
+
+	connectedCallback() {
+		// Set default styles.
+		this.attachShadow( { mode: 'open' } );
+		this.shadowRoot.innerHTML = `
+			<style>
+				:host {
+					display: block;
+				}
+			</style>
+			<slot></slot>
+		`;
+
+		// Update the component content.
+		this.setContent();
+
+		// Attach a listener to update the content when audiences are changed.
+		window.Altis.Analytics.on( 'updateAudiences', this.setContent );
+	}
+
+	setContent() {
+		const audiences = window.Altis.Analytics.getAudiences();
+
+		// Get associated templates.
+		const templates = document.querySelectorAll( `template[data-parent-id="${ this.clientId }"]` );
```

Comment:
Could this be scoped to the current Experience Block HTML element, or are they somewhere else in the DOM?

Reply:
They're just above it in the DOM, with web-components any tags you place insidethat you don't want to be overridden need to go in the shadow DOM which you can't really do with direct HTML output. You'd have to listen to an event within the component for tags being rendered inside it and pull them out into the shadow DOM at that point which is a pain to do so it's much easier to output them above, that way you know they're definitely available.

Reply:
Ahh ok -- wondering if it would be worth looking at the previous sibling? Just wondering if there's overhead to `document.querySelectorAll` a lot

Reply:
It's pretty quick I think, and using the attribute selector makes it pretty specific. This would be a micro-optimisation at this point anyway so we can come back to it.